File: src/main/java/com/example/cqrs/command/dto/WithdrawRequest.java
package com.example.cqrs.command.dto;

import lombok.*;

@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
@Getter
public class WithdrawRequest {

    private String accountId;
    private double amount;
    private String userId;

    // factory method
    public static WithdrawRequest of(String accountId, double amount, String userId) {
        return new WithdrawRequest(accountId, amount, userId);
    }

}

File: src/main/java/com/example/cqrs/command/dto/WithdrawRequest.kt
package com.example.cqrs.command.dto

data class WithdrawRequest(
    val accountId: String,
    val amount: Double,
    val userId: String
) {
    companion object {
        fun of(accountId: String, amount: Double, userId: String): WithdrawRequest {
            return WithdrawRequest(accountId, amount, userId)
        }
    }
}
